powersimdata/scenario/scenario.py

```diff
@@ -41,6 +42,15 @@ def __init__(self, descriptor):
             except AttributeError:
                 return
 
+    def __getattr__(self, name):
```

Comment:
Any concern about this breaking calls to `getattr`? One similar idea would be to do a bunch of `setattr` calls in the constructor for the exported methods, which would leave the default behavior here, and make the methods available for tab completion on the scenario object (not sure if this is the case here).

Reply:
From my limited research, the `__getattribute__` method is called any time you look up an attribute, and the `__getattr__` method is called only if `__getattribute__` fails.

What's an example of a call that could break here?

Reply:
Oh I see, not too worried in this case. I'm not sure if it's used by automated tools though, like pytest or sphinx, but we can see what happens. I still think it'd be nice to have the attributes set up front, so tab complete or `dir(scenario)` would have them, but we'd need to refresh when the state changes. What do you think?

Reply:
Could we define `__dir__` so that it runs `super().__dir__()` and then adds `self.state.exported_methods`?

Reply:
@danielolsen I think that should work.
```
def __dir___(self):
return sorted(set(super().__dir__() + self.state.exported_methods))
```

Reply:
How your code would look like @jon-hagg with the `setattr`?

Reply:
Nevermind that, I think overriding `__dir__` achieves the same thing with less work. Bainan's snippet works but need to convert the exported_methods to a list, or something similar.

Reply:
Done.